tests/integration/test_utils/data_source_config/snowflake.py

```diff
@@ -106,33 +107,56 @@ def make_batch(self) -> Batch:
 
     @override
     def setup(self) -> None:
-        column_types: Dict[str, SnowflakeType] = self.get_column_types()
-        columns: List[Column] = [
-            Column(name, column_type) for name, column_type in column_types.items()
-        ]
-        self.table = Table(
-            self.table_name,
-            self.metadata,
-            *columns,
-            schema=self.snowflake_connection_config.SNOWFLAKE_SCHEMA,
-        )
+        main_table = self._create_table(name=self.table_name, columns=self.get_column_types())
+        extra_tables = {
+            table_name: self._create_table(
+                name=table_name,
+                columns=self.get_extra_column_types(table_name),
+            )
+            for table_name in self.extra_data
+        }
+        self.tables = [main_table, *extra_tables.values()]
+
         self.metadata.create_all(self.engine)
         with self.engine.connect() as conn:
             # pd.DataFrame(...).to_dict("index") returns a dictionary where the keys are the row
             # index and the values are a dict of column names mapped to column values.
             # Then we pass that list of dicts in as parameters to our insert statement.
             #   INSERT INTO test_table (my_int_column, my_str_column) VALUES (?, ?)
             #   [...] [('1', 'foo'), ('2', 'bar')]
-            conn.execute(insert(self.table), list(self.data.to_dict("index").values()))
-            conn.commit()
+            with conn.begin():
+                conn.execute(insert(main_table), list(self.data.to_dict("index").values()))
+                for table_name, table_data in self.extra_data.items():
+                    conn.execute(
+                        insert(extra_tables[table_name]),
+                        list(table_data.to_dict("index").values()),
+                    )
 
     @override
     def teardown(self) -> None:
-        if self.table is not None:
-            self.table.drop(self.engine)
+        if self.tables:
+            for table in self.tables:
+                table.drop(self.engine)
+
+    def _create_table(self, name: str, columns: Mapping[str, Any]) -> Table:
+        column_list: list[Column] = [
+            Column(col_name, col_type) for col_name, col_type in columns.items()
+        ]
+        return Table(
+            name,
+            self.metadata,
+            *column_list,
+            schema=self.snowflake_connection_config.SNOWFLAKE_SCHEMA,
+        )
 
-    def get_column_types(self) -> Dict[str, _SnowflakeType]:
```

Comment:
Similar to the comment at the top of this file, the `_SnowflakeType` type arg was providing no type safety because it wasn't bound to anything. We don't run it in CI, but pyright complained about this. Switching to `Any` looks worse, but provides the same lack of typing. We need to circle back on better typing here.